configure.ac

```diff
@@ -2621,12 +2621,22 @@ fi
 LIB_SOCKET_NSL
 AX_HARDEN_CC_COMPILER_FLAGS
 
-# link to ws2_32 if on mingw
+# if mingw then link to ws2_32 for sockets
 case $host_os in
-    *mingw32)
+    mingw*)
         LDFLAGS="$LDFLAGS -lws2_32" ;;
 esac
 
+# add wolfSSL defines for shared/static
+if test "$enable_shared" = "yes"
+then
+    AC_DEFINE([WOLFSSL_SHARED], [1], [Define if shared library avialble.])
+fi
+if test "$enable_static" = "yes"
+then
+    AC_DEFINE([WOLFSSL_STATIC], [1], [Define if static library avialble.])
```

Comment:
I suspect that pushing both WOLFSSL_SHARED and WOLFSSL_STATIC (at the same time by default) into config.h is going to break the visibility.  That is, when building the static version of the library WOLFSSL_SHARED will still be defined and dll imports/exports will be incorrect.  Was this tested?